### PR TITLE
usm: http2: process http2 marker

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?
process http2 marker if presents.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/ebpf/c/protocols/http2/decoding.h [hunk 1]
```diff
@@ -305,6 +306,16 @@ static __always_inline void process_headers_frame(struct __sk_buff *skb, http2_s
 }
 
 static __always_inline bool http2_entrypoint(struct __sk_buff *skb, skb_info_t *skb_info, conn_tuple_t *tup, http2_ctx_t *http2_ctx) {
+    // Checking for preface
+    if (skb_info->data_off + HTTP2_MARKER_SIZE <= skb->len) {
+        char preface[HTTP2_MARKER_SIZE];
+        bpf_memset((char*)preface, 0, HTTP2_MARKER_SIZE);
```
Comment (Yumasi) [inline pkg/network/ebpf/c/protocols/http2/decoding.h:312]: Is this memset needed since the entire buffer is filled just after with `bpf_skb_load_bytes`?

**Review by Yumasi — CHANGES_REQUESTED** [review]
Just a small comment, other than that it looks good. Maybe add some details about the leak being fixed here in the description or ticket.

File: pkg/network/ebpf/c/protocols/http2/decoding.h (hunk 1 above)
Comment (guyarb) [reply pkg/network/ebpf/c/protocols/http2/decoding.h:312]: it is a best practice, is it required? honestly I'm not sure. I do remember having troubles in the past with the verifier wasn't happy on uninitialized variables, but not sure if that's the case
anyhow, if you're right, then my suggestion would be to make it in a different PR and fixing it for the entire usm codebase
WDYT?

Comment (Yumasi) [reply pkg/network/ebpf/c/protocols/http2/decoding.h:312]: I think it is best practice to fully initialise a memory region before reading into it, but not necessarily with `memset`. Modern compilers will remove calls to `memset` if they are followed by a `memcpy`, which, by looking at [its implementation](https://elixir.bootlin.com/linux/v4.7/source/net/core/filter.c#L1423), is basically what `bpf_skb_load_bytes` is.

Regarding the verifier, I was able to use `skb_load_bytes` to write into a stack allocated buffer, without having to `memset` it first, when implementing the gRPC classification, so I think it is fine with it now.

> my suggestion would be to make it in a different PR and fixing it for the entire usm codebase

Sounds good to me!

**Review by Yumasi — APPROVED** [review]

**Pull request merged**